controllers/microservices_controller.go

```diff
@@ -31,16 +39,72 @@ type MicroservicesReconciler struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.12.2/pkg/reconcile
 func (r *MicroservicesReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	log := log.FromContext(ctx)
+	log = log.WithValues("tempo", req.NamespacedName)
+	tempo := v1alpha1.Microservices{}
+	if err := r.Get(ctx, req.NamespacedName, &tempo); err != nil {
+		if !apierrors.IsNotFound(err) {
+			log.Error(err, "unable to fetch TempoMicroservices")
+		}
+
+		// we'll ignore not-found errors, since they can't be fixed by an immediate
+		// requeue (we'll need to wait for a new notification), and we can get them
+		// on deleted requests.
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	objects, err := manifests.BuildAll(manifests.Params{Tempo: tempo})
+	// TODO(pavolloffay) check error type and change return appropriately
+	if err != nil {
+		return ctrl.Result{
+			Requeue:      true,
+			RequeueAfter: time.Second,
+		}, err
+	}
+	errCount := 0
+	for _, obj := range objects {
+		l := log.WithValues(
+			"object_name", obj.GetName(),
+			"object_kind", obj.GetObjectKind(),
+		)
+
+		if isNamespaceScoped(obj) {
+			obj.SetNamespace(req.Namespace)
+			if err := ctrl.SetControllerReference(&tempo, obj, r.Scheme); err != nil {
+				l.Error(err, "failed to set controller owner reference to resource")
+				errCount++
+				continue
+			}
+		}
 
-	// TODO(user): your logic here
+		desired := obj.DeepCopyObject().(client.Object)
+		mutateFn := manifests.MutateFuncFor(obj, desired)
+
+		op, err := ctrl.CreateOrUpdate(ctx, r.Client, obj, mutateFn)
+		if err != nil {
+			l.Error(err, "failed to configure resource")
+			errCount++
+			continue
+		}
+
+		l.Info(fmt.Sprintf("Resource has been %s", op))
+	}
 
 	return ctrl.Result{}, nil
 }
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *MicroservicesReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
-		For(&tempov1alpha1.Microservices{}).
+		For(&v1alpha1.Microservices{}).
 		Complete(r)
 }
+
+func isNamespaceScoped(obj client.Object) bool {
+	switch obj.(type) {
+	case *rbacv1.ClusterRole, *rbacv1.ClusterRoleBinding:
```

Comment:
Isnt there another way to figure out what objects are namespace scoped? Its just a feeling, but `isNamespaceScoped` does only consider 2 types and feels misleading.

Reply:
Do you mean to figure out the scope from the k8s API? 

I don't think there is a way.